pkg/portal/cluster/statistics.go

```diff
@@ -0,0 +1,115 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"time"
+
+	prometheusAPI "github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+)
+
+type MetricValue struct {
+	Timestamp time.Time `json:"timestamp"`
+	Value     float64   `json:"value"`
+}
+
+type Metrics struct {
+	Name  string        `json:"metricname"`
+	Value []MetricValue `json:"metricvalue"`
+}
+
+func (c *client) Statistics(ctx context.Context, httpClient *http.Client, promQuery string, duration string, endTime time.Time) ([]Metrics, error) {
+	return c.fetcher.Statistics(ctx, httpClient, promQuery, duration, endTime)
+}
+
+func (f *realFetcher) Statistics(ctx context.Context, httpClient *http.Client, promQuery string, duration string, endTime time.Time) ([]Metrics, error) {
+	promConfig := prometheusAPI.Config{
+		Address: "http://prometheus-k8s-0:9090",
+		Client:  httpClient,
+	}
+
+	client, err := prometheusAPI.NewClient(promConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	v1api := v1.NewAPI(client)
+
+	startTime, err := getStartTimeFromDuration(duration, endTime)
+	if err != nil {
+		return nil, err
+	}
+	value, warning, err := v1api.QueryRange(ctx, promQuery, v1.Range{
+		Start: startTime,
+		End:   endTime,
+		Step:  time.Minute * 2,
+	})
+	if len(warning) > 0 {
+		f.log.Warn(warning)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	valueMatrix := value.(model.Matrix)
+	return convertToStringMap(valueMatrix), nil
+}
+
+func convertToStringMap(v model.Matrix) []Metrics {
```

Comment:
I'm confused, this doesn't convert it to a stringmap does it?

Reply:
no .. yaa that name is consufing, I will change that name.

Reply:
name changed resolving this conversation.